packages/nox/test/tpetra/MultiVectorOpsTests.cpp

```diff
@@ -0,0 +1,488 @@
+// @HEADER
+// *****************************************************************************
+//            NOX: An Object-Oriented Nonlinear Solver Package
+//
+// Copyright 2002 NTESS and the NOX contributors.
+// SPDX-License-Identifier: BSD-3-Clause
+// *****************************************************************************
+// @HEADER
+
+#include "Tpetra_Core.hpp"
+#include "Tpetra_Map.hpp"
+#include "Tpetra_MultiVector.hpp"
+#include "Tpetra_Vector.hpp"
+
+#include "NOX.H"
+#include "NOX_Tpetra_MultiVector.hpp"
+#include "NOX_Tpetra_Vector.hpp"
+
+#include "Teuchos_UnitTestHarness.hpp"
+
+template <typename LocalOrdinal, typename GlobalOrdinal, typename Node>
+using tpetra_map_t = ::Tpetra::Map<LocalOrdinal, GlobalOrdinal, Node>;
+
+template <typename Scalar, typename LocalOrdinal, typename GlobalOrdinal, typename Node>
+using tpetra_vector_t = ::Tpetra::Vector<Scalar, LocalOrdinal, GlobalOrdinal, Node>;
+
+template <typename Scalar, typename LocalOrdinal, typename GlobalOrdinal, typename Node>
+using tpetra_multivector_t = ::Tpetra::MultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node>;
+
+template <typename Scalar, typename LocalOrdinal, typename GlobalOrdinal, typename Node>
+using nox_tpetra_vector_t = NOX::Tpetra::Vector<Scalar, LocalOrdinal, GlobalOrdinal, Node>;
+
+template <typename Scalar, typename LocalOrdinal, typename GlobalOrdinal, typename Node>
+using nox_tpetra_multivector_t = NOX::Tpetra::MultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node>;
+
+template <typename Scalar>
+using magnitude_type = typename Teuchos::ScalarTraits<Scalar>::magnitudeType;
+
+//! Function to check solution.
+template <typename Scalar, typename LocalOrdinal, typename GlobalOrdinal, typename Node>
+bool checkMultiVectors(
+    const Teuchos::RCP<tpetra_multivector_t<Scalar, LocalOrdinal, GlobalOrdinal, Node>>& a,
+    const Teuchos::RCP<tpetra_multivector_t<Scalar, LocalOrdinal, GlobalOrdinal, Node>>& b,
+    const Kokkos::View<magnitude_type<Scalar>*, Kokkos::HostSpace>& expt,
+    Teuchos::FancyOStream& out,
+    bool success
+)
+{
+    constexpr magnitude_type<Scalar> tol = 1.0e-14;
```

Comment:
This assumes a double is the scalar type. Maybe use something like: 

constexpr magnitude_type<Scalar> tol = 10.0*std::numeric_limits<Scalar>::epsilon();

Reply:
OK. That's a good suggestion indeed. I've used the new `Kokkos_NumericLimits.hpp`.